cache/refs.go

```diff
@@ -307,10 +282,8 @@ func (cr *cacheRecord) remove(ctx context.Context, removeSnapshot bool) error {
 			return err
 		}
 	}
-	if removeSnapshot {
-		if err := cr.cm.LeaseManager.Delete(context.TODO(), leases.Lease{ID: cr.ID()}); err != nil {
-			return errors.Wrapf(err, "failed to remove %s", cr.ID())
-		}
+	if err := cr.cm.LeaseManager.Delete(context.TODO(), leases.Lease{ID: cr.ID()}); err != nil {
+		return errors.Wrapf(err, "failed to remove %s", cr.ID())
```

Comment:
Yeah I believe it should because this line isn't deleting a snapshot resource from the lease, it's deleting the entire lease itself (which inherently deletes all the resources associated with the lease, right?). So even though the snapshot ID can change depending on if the ref was finalized, the lease's ID will always just be the ref's ID.

Reply:
Thank you for comfirming this. I think you're right.